### PR TITLE
Topic/union converter

### PR DESCRIPTION
**Pull request by mkorpela**
Union conversion.
Depends on https://github.com/robotframework/robotframework/pull/3713.

### CODE REVIEW

**Comment by mkorpela** [comment]
Todolist:

- [x] string "None" is not converted by default
- [x] Union is handled
- [x] Types inside union are matched even if there is no converter for the specific type
- [ ] float -> int converter
- [ ] int -> float converter
- [ ] number -> timedelta converter
- [ ] testing

File: atest/testdata/keywords/type_conversion/unions.robot [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    Union of int float and string   ${-110}  ${-110}
+
+Custom type inside of union
+    ${myobject}=  Create my object
+    Custom type in union  myobject     <class 'str'>
+    Custom type in union  ${myobject}  <class 'unions.MyObject'>
+
+Unexpected object is just passed when in union
+    ${object}=  Create unexpected object
+    Custom type in union  ${object}     <class 'unions.UnexpectedObject'>
```
Comment (mkorpela) [inline atest/testdata/keywords/type_conversion/unions.robot:29]: I'm really unsure how this should really work. I would prefer an exception but now it passes.

**Comment by mkorpela** [comment]
This also has enum None conversion fix.

**Review by pekkaklarck — APPROVED** [review]
The general approach looks. Only functional thing I want to change is how unrecognized types are handled. When used like `arg: MyThing`, we don't do any conversion and just pass the value as-is. For consistency reasons `arg: Union[int, MyThing]` shouldn't cause an error and instead values that cannot be converted to an integer should be passed as-is. I'll change that logic after merging this PR and may do some fine-tuning/cleanup also otherwise.

**Pull request merged**